### PR TITLE
Add analyse slither and github action

### PR DESCRIPTION
**Pull request by gfournieriExec**
Left 
Summary
 - [uninitialized-state](#uninitialized-state) (16 results) (High)
 - [reentrancy-no-eth](#reentrancy-no-eth) (2 results) (Medium)
 - [reentrancy-benign](#reentrancy-benign) (2 results) (Low)
 - [reentrancy-events](#reentrancy-events) (4 results) (Low)
 - [unused-import](#unused-import) (5 results) (Informational)

### CODE REVIEW

File: contracts/modules/delegates/SignatureVerifier.v8.sol [hunk 1]
```diff
@@ -59,14 +59,17 @@ contract SignatureVerifier is DelegateBase {
         bytes32 messageHash,
         bytes calldata signature
     ) internal view returns (bool) {
-        address recoveredAddress;
+        address recoveredAddress = address(0); // Initialize local variable
         if (signature.length == 65) {
+            //slither-disable-start unused-return
+            // Ignoring r and s as they are not needed
```
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply contracts/modules/delegates/SignatureVerifier.v8.sol:65]: It's just an explanation of the "unused-return"

File: contracts/modules/delegates/IexecPoco2Delegate.sol [hunk 2]
```diff
@@ -510,13 +510,18 @@ contract IexecPoco2Delegate is IexecPoco2, DelegateBase, IexecEscrow, SignatureV
             // }
 
             // Pre solidity 0.6.0 version
+            //slither-disable-start low-level-calls
```
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply contracts/modules/delegates/IexecPoco2Delegate.sol:513]: Will try it

File: contracts/tools/testing/slither/Slither.sol [hunk 3]
```diff
@@ -13,4 +13,6 @@ import {IexecPoco2Delegate} from "../../../modules/delegates/IexecPoco2Delegate.
  * @dev This contract aggregates multiple delegate contracts into a single entry point
  *      to facilitate static analysis using Slither.
  */
+//slither-disable-start unused-state
 contract Slither is IexecPocoAccessorsDelegate, IexecPoco1Delegate, IexecPoco2Delegate {}
```
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply contracts/tools/testing/slither/Slither.sol:17]: I don't understand why ? These contracts are also used in github action

File: contracts/modules/delegates/SignatureVerifier.v8.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyjams) [reply contracts/modules/delegates/SignatureVerifier.v8.sol:65]: Return values are `(recoveredAddress, error, signatureLength)`, why mentioning  r & s?

File: contracts/tools/testing/slither/Slither.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeremyjams) [reply contracts/tools/testing/slither/Slither.sol:17]: Ok, you are sure behavior is the same with the Github action ? (I guess but not 100% sure!)

[the remark this answers is not in the document]
Comment (gfournieriExec) [reply contracts/tools/testing/slither/Slither.sol:17]: yes works the same

File: contracts/modules/delegates/IexecPoco2Delegate.sol [hunk 4]
```diff
@@ -510,13 +510,18 @@ contract IexecPoco2Delegate is IexecPoco2, DelegateBase, IexecEscrow, SignatureV
...
+            // See Halborn audit report for details
```
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply contracts/modules/delegates/IexecPoco2Delegate.sol:513]: Done

File: contracts/modules/delegates/IexecPoco2Delegate.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply contracts/modules/delegates/IexecPoco2Delegate.sol]: Works yes perfect thanks

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -21,4 +21,17 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           slug: iExecBlockchainComputing/PoCo
-          
+      - name: Run static analysis with slither
+        uses: crytic/slither-action@v0.4.0
+        id: slither
+        with:
+          target: "contracts/tools/testing/slither/"
+          solc-version: '0.8.21'
```
Comment (jeremyjams) [inline .github/workflows/main.yml:29]:
```suggestion
```

File: contracts/modules/delegates/IexecPoco1Delegate.sol [hunk 6]
```diff
@@ -232,6 +232,7 @@ contract IexecPoco1Delegate is
         /**
          * Check orders authenticity
          */
+        //slither-disable-next-line uninitialized-local
```
Comment (jeremyjams) [inline contracts/modules/delegates/IexecPoco1Delegate.sol:235]: Not sure what could happen here

File: contracts/tools/testing/slither/Slither.sol [hunk 7]
```diff
@@ -13,4 +13,6 @@ import {IexecPoco2Delegate} from "../../../modules/delegates/IexecPoco2Delegate.
  * @dev This contract aggregates multiple delegate contracts into a single entry point
  *      to facilitate static analysis using Slither.
  */
+//slither-disable-start unused-state
```
Comment (jeremyjams) [inline contracts/tools/testing/slither/Slither.sol:16]: Not sure to understand why we need that annotation :thinking:

File: contracts/modules/delegates/IexecPoco2Delegate.sol [hunk 8]
```diff
@@ -510,12 +510,16 @@ contract IexecPoco2Delegate is IexecPoco2, DelegateBase, IexecEscrow, SignatureV
...
+            //slither-disable-next-line low-level-calls
             (bool success, bytes memory returndata) = target.call{gas: m_callbackgas}(
                 abi.encodeWithSignature("receiveResult(bytes32,bytes)", _taskid, _resultsCallback)
             );
             assert(gasleft() > m_callbackgas / 63);
             // silent unused variable warning
+            //slither-disable-next-line redundant-statements
             success;
+            //slither-disable-next-line redundant-statements
```
Comment (zguesmi) [inline contracts/modules/delegates/IexecPoco2Delegate.sol:522]: Can we remove these variables and ignore solidity warning `Return value of low-level calls not used`

File: .github/workflows/main.yml (hunk 5 above)
Comment (gfournieriExec) [reply .github/workflows/main.yml:29]: Needed indeed : 
[-] SOLCVER was not set; guessing.
[-] Guessed 0.8.0.
Installing solc '0.8.0'...
Version '0.8.0' installed.

=> Error: Source file requires different compiler version (current compiler is 0.8.0+commit.c7dfd78e.Linux.g++) - note that nightly builds are considered to be strictly less than the released version
 --> node_modules/@openzeppelin/contracts-v5/interfaces/IERC5313.sol:4:1:
  |
4 | pragma solidity ^0.8.20;

File: contracts/modules/delegates/IexecPoco1Delegate.sol (hunk 6 above)
Comment (gfournieriExec) [reply contracts/modules/delegates/IexecPoco1Delegate.sol:235]: Why do mean ? Slither reports the fact that `Matching memory ids;` is not initialized

File: contracts/modules/delegates/IexecPoco2Delegate.sol (hunk 8 above)
Comment (gfournieriExec) [reply contracts/modules/delegates/IexecPoco2Delegate.sol:522]: We can, but I did not wanted to modify our poco 1 / 2 delegate contracts

File: contracts/tools/testing/slither/Slither.sol (hunk 7 above)
Comment (gfournieriExec) [reply contracts/tools/testing/slither/Slither.sol:16]: ` - [unused-state](#unused-state) (25 results) (Informational)`
`ERC1538Store.m_funcs (contracts/Store.v8.sol#38) is never used in Slither (contracts/tools/testing/slither/Slither.sol#17)`

Comment (gfournieriExec) [reply contracts/tools/testing/slither/Slither.sol:16]: It just say that a lot of variable are not used in this specifc contract `Slither.sol` wich is normal 
So I prefer skip the slither check to have less verbosity in the report

File: contracts/Store.v8.sol [hunk 9]
```diff
@@ -54,15 +54,23 @@ interface IRegistry is IERC721Enumerable {
  */
 abstract contract Store is ERC1538Store {
     // Registries
+    //slither-disable-next-line constable-states
     IRegistry internal m_appregistry;
+    //slither-disable-next-line constable-states
     IRegistry internal m_datasetregistry;
+    //slither-disable-next-line constable-states
```
[the remark this answers is not in the document]
Comment (gfournieriExec) [reply contracts/Store.v8.sol:59]: See this comment https://github.com/iExecBlockchainComputing/PoCo/pull/90#discussion_r1637683393
I'm open for discussion

File: .github/workflows/main.yml (hunk 5 above)
Comment (jeremyjams) [reply .github/workflows/main.yml:29]: OK thanks :+1:

File: contracts/Store.v8.sol (hunk 9 above)
[the remark this answers is not in the document]
Comment (jeremyjams) [reply contracts/Store.v8.sol:59]: Generally it looks safer to add unitary annotations to be sure to not introduce not-seen bugs if future code is added within the block

File: contracts/modules/delegates/IexecPoco1Delegate.sol (hunk 6 above)
Comment (jeremyjams) [reply contracts/modules/delegates/IexecPoco1Delegate.sol:235]: Yes, I don't see here if something bad could happen here

Comment (gfournieriExec) [reply contracts/modules/delegates/IexecPoco1Delegate.sol:235]: Here nothing

**Review by jeremyjams — APPROVED** [review]
Thank you!

**Review by zguesmi — APPROVED** [review]
Good!

**Pull request merged**